metadata/moa/scripts/nbconverted/2.map-broad_id.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import pandas as pd
+import numpy as np
+import rdkit.Chem.inchi as inchi
+
+
+# Sample files from each version of the repurposing hub are read
+
+# In[2]:
+
+
+samples_2017 = pd.read_csv('clue/repurposing_samples_20170327.txt',usecols=['broad_id','pert_iname','InChIKey'],delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2018a = pd.read_csv('clue/repurposing_samples_20180516.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2018b = pd.read_csv('clue/repurposing_samples_20180907.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+samples_2020 = pd.read_csv('clue/repurposing_samples_20200324.txt', usecols=['broad_id','pert_iname','InChIKey','deprecated_broad_id'], delimiter='\t', comment='!', encoding='iso-8859-1')
+
+
+# In[3]:
+
+
+# 2017 version is missing deprecated_broad_id
+samples_2017['deprecated_broad_id'] = np.nan
```

Comment:
I am a fan of adding columns with [`.assign()`](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.assign.html). I thought I remembered readings somewhere that assign is the preferred method, but I can't find the resource now...

Reply:
Thanks for the suggestion! [Looks like it is good way](https://stackoverflow.com/questions/12555323/adding-new-column-to-existing-dataframe-in-python-pandas) to avoid `SettingWithCopyWarning`